src/osd/OSD.cc

```diff
@@ -6642,7 +6642,7 @@ void OSD::handle_osd_map(MOSDMap *m)
       OSDMap *o = new OSDMap;
       if (e > 1) {
 	bufferlist obl;
-	get_map_bl(e - 1, obl);
+	assert(get_map_bl(e - 1, obl));
```

Comment:
please assert on a different line.. we don't want any side-effects in the assert itself